docs/listings-apis

```diff
@@ -0,0 +1,123 @@
+# Listings API
+
+## Overview
+
+This API enables interaction with a listings data source using JSON format, providing CRUD (Create, Read, Update, Delete) operations for managing listings. Each endpoint interacts with `listings.json` and performs operations such as retrieval, addition, updates, and deletion of listing data. JSON stubs serve as examples for request and response data structures.
+
+## Endpoints
+
+### GET /api/listings
+
+Retrieve all listings available.
+
+#### Example Response
+```json
+{
+  "listings": [
+    {
+      "id": 1,
+      "author": "nameExample",
+      "title": "Used Bicycle",
+      "price": 150.00,
+      "description": "A gently used mountain bike.",
+      "seller": "Student1",
+      "sellerID": 294720,
+      "class": "Sports",
+      "condition": "Used",
+      "saleAvailability": "Available"
+    }
+  ]
+}
+```
+
+---
+
+### GET /api/listings/{listingId}
+
+Retrieve a specific listing by its ID.
+
+#### Example Request
+`GET /api/listings/1`
+
+#### Example Response
+```json
+{
+  "id": 1,
+  "author": "nameExample",
+  "title": "Used Bicycle",
+  "price": 150.00,
+  "description": "A gently used mountain bike.",
+  "seller": "Student1",
+  "sellerID": 294720,
+  "class": "Sports",
+  "condition": "Used",
+  "saleAvailability": "Available"
+}
+```
+
+---
+
+### POST /api/listings
+
+Create a new listing. The request body should contain the listing details.
+
+#### Example Request Body
+```json
+{
+  "author": "nameExample",
+  "title": "New Item Title",
+  "price": 200.00,
+  "description": "Description of the new item.",
+  "seller": "Student3",
+  "sellerID": 294722,
```

Comment:
We will read the user's session cookie and get the corresponding `user` from the DB, so no need to provide the `seller` or `sellerId` in the request. This can be done in a later commit